src/utilities/alfalfa/AlfalfaComponent.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef ALFALFA_COMPONENT_HPP
+#define ALFALFA_COMPONENT_HPP
+
+#include <json/json.h>
+
+#include "../UtilitiesAPI.hpp"
+
+namespace openstudio {
+namespace alfalfa {
+  enum Capability
+  {
+    Input = 1,
+    Output = 2
+  };
+
+  inline Capability operator|(const Capability& a, const Capability& b) {
+    return static_cast<Capability>(static_cast<int>(a) | static_cast<int>(b));
+  }
+
+  inline Capability operator&(const Capability& a, const Capability& b) {
+    return static_cast<Capability>(static_cast<int>(a) & static_cast<int>(b));
+  }
+
+  class UTILITIES_API AlfalfaComponent
+  {
+   public:
+    AlfalfaComponent(const std::string& type, const Capability capabilities);
```

Comment:
Should this be public? I'm getting hints of "this is just a base class and you're supposed to only instantiate the derived ones"

Reply:
That is a good question, it could just as easily be protected without requiring any change, right?